Applications/web_portal/web_app/library/source/pages/login_page/login_page.tsx

```diff
@@ -11,15 +15,220 @@ export interface Properties {
   constructor(properties: Properties) {
     super(properties);
+    this.state = {
+      errorMessage: null
+    };
+    this.onLogin = this.onLogin.bind(this);
+    this.onInputChange = this.onInputChange.bind(this);
+    this.onKeyDown = this.onKeyDown.bind(this);
+  }
+
+  public componentDidMount(): void {
+    window.addEventListener('keydown', this.onKeyDown);
+  }
+
+  public componentWillUnmount(): void {
+    window.removeEventListener('keydown', this.onKeyDown);
   }
 
   public render(): JSX.Element {
-    return <div>Login Page</div>;
+    return (
+      <Center width='100%' height='100%'
+          className={css(LoginPage.STYLE.page)}>
+        <HBoxLayout width='320px' height='462px'>
+          <Padding size='18px'/>
+          <VBoxLayout width='284px' height='462px'>
+            <Padding size='60px'/>
+            <HBoxLayout width='100%' height='50px'>
+              <Padding/>
+              <object id='static-logo'
+                data='resources/login_page/logo-static.svg'
+                type='image/svg+xml' className={
+                css(LoginPage.STYLE.logoVisible)}/>
+              <object id='animated-logo'
+                data='resources/login_page/logo-animated.svg'
+                type='image/svg+xml' className={
+                css(LoginPage.STYLE.logoInvisible)}>
+              </object>
+              <Padding/>
+           </HBoxLayout>
+            <Padding size='60px'/>
+            <input type='text' placeholder='Username' autoComplete='off'
+              className={css(LoginPage.STYLE.inputBox)}
+              onChange={() => this.onInputChange()}
+              onFocus={() => this.usernameInputField.placeholder = ''}
+              onBlur={() => this.usernameInputField.placeholder = 'Username'}
+              ref={((ref) => this.usernameInputField = ref)}/>
+            <Padding size='20px'/>
+            <input type='password' placeholder='Password' autoComplete='off'
+              className={css(LoginPage.STYLE.inputBox)}
+              onChange={() => this.onInputChange()}
+              onFocus={() => this.passwordInputField.placeholder = ''}
+              onBlur={() => this.passwordInputField.placeholder = 'Password'}
+              ref={((ref) => this.passwordInputField = ref)}/>
+            <Padding size='50px'/>
+            <button className={css(LoginPage.STYLE.signInButton)}
+                onClick={this.onLogin}
+                ref={((ref) => this.submitButton = ref)}>
+              Sign In
+            </button>
+            <Padding size='30px'/>
+            <span className={css(LoginPage.STYLE.errorMessage)}>
+              {this.state.errorMessage}
+            </span>
+            <Padding size='60px'/>
+          </VBoxLayout>
+          <Padding size='18px'/>
+        </HBoxLayout>
+      </Center>);
+  }
+
+  private async onLogin() {
+    if(this.usernameInputField.value.trim() !== '') {
+      const staticLogo = document.getElementById('static-logo');
```

Comment:
Use refs to capture elements instead of using getElementById. Make two private fields of type HTMLObjectElement and then assign them using ref=...